core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -78,11 +78,11 @@ object ByteBufferMessageSet {
   }
 
   /** Deep iterator that decompresses the message sets and adjusts timestamp and offset if needed. */
-  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): Iterator[MessageAndOffset] = {
+  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): CloseableIterator[MessageAndOffset] = {
```

Comment:
@junrao, is this method public API for people who are still using the old clients? Or is it OK to change the method signature? If it's not, then we need to introduce a new method and deprecate this one.